src/bosh_azure_cpi/lib/cloud/azure/restapi/vmss_rest.rb

```diff
@@ -0,0 +1,318 @@
+                'name' => "ipcfg-#{index}",
+                'properties' => {
+                  'subnet' => {
+                    'id' => network[:subnet][:id]
+                  },
+                  'loadBalancerInboundNatPools' => [],
+                  'loadBalancerBackendAddressPools' => []
+                }
+              }
+            ]
+          }
+        }
+        unless vm_params[:load_balancer].nil?
+          nic['properties']['ipConfigurations'][0]['properties']['loadBalancerBackendAddressPools'].push(
+            'id' => vm_params[:load_balancer][:backend_address_pools][0][:id]
+          )
+        end
+
+        unless network[:network_security_group].nil?
+          nic['properties']['networkSecurityGroup'] = {
+            'id' => network[:network_security_group][:id]
+          }
+        end
+        network_profile['networkInterfaceConfigurations'].push(nic)
+      end
+
+      os_disk = {
+        'createOption' => 'FromImage',
+        'caching'      => vm_params[:os_disk][:disk_caching],
+        'managedDisk' => {
+          'storageAccountType' => 'Standard_LRS'
+        }
+      }
+      # TODO: investigate whether vmss support overwrite diskSizeGB.
+      # os_disk['diskSizeGB'] = vm_params[:os_disk][:disk_size] unless vm_params[:os_disk][:disk_size].nil?
+      data_disks = []
+      unless vm_params[:ephemeral_disk].nil?
+        data_disks.push(
+          'createOption' => 'Empty',
+          'diskSizeGB'   => vm_params[:ephemeral_disk][:disk_size],
+          'lun'          => EPHEMERAL_DISK_LUN
+        )
+      end
+      storage_profile = {
+        'imageReference' => {
+          'id' => vm_params[:image_id]
+        },
+        'osDisk' => os_disk,
+        'dataDisks' => data_disks
+      }
+
+      vmss = {
+        'sku' => {
+          'tier' => 'Standard',
+          'capacity' => vm_params[:initial_capacity],
+          'name' => vm_params[:instance_type]
+        },
+        'location' => vm_params[:location],
+        'properties' => {
+          'overprovision' => false,
+          'virtualMachineProfile' => {
+            'storageProfile' => storage_profile,
+            'osProfile' => os_profile,
+            'networkProfile' => network_profile
+          },
+          'upgradePolicy' => {
+            'mode' => 'Manual'
+          }
+        },
+        'zones' => vm_params[:availability_zones]
+      }
+
+      params = {
+        'validating' => 'true'
+      }
+      http_put(url, vmss, params)
+    end
+
+    def scale_vmss_up(resource_group_name, vmss_name, number)
+      url = rest_api_url(REST_API_PROVIDER_COMPUTE, REST_API_VIRTUAL_MACHINE_SCALE_SETS, resource_group_name: resource_group_name, name: vmss_name)
+      vmss = _get_vmss_by_name(resource_group_name, vmss_name)
+      vmss['sku']['capacity'] = vmss['sku']['capacity'].to_i + number
+      params = {
+        'validating' => 'true'
+      }
+      http_put(url, vmss, params)
+    end
+    # VMSS related ends
```

Comment:
Remove this comment?

Reply:
this is a big wrapper, it's paired to the 
# VMSS Instance starts
# VMSS Instance ends.
# VMSS related starts
# VMSS related ends

Reply:
Just thinking the comment is not very useful. Anyway, I'm fine to keep them.